src/components/Photography/PhotographyGridSection.tsx

```diff
@@ -3,148 +3,141 @@ import 'react-image-lightbox/style.css'
 import _ from 'lodash'
 import fp from 'lodash/fp'
 import { DateTime } from 'luxon'
-import React, { Component } from 'react'
+import React, { useState } from 'react'
 import Lightbox from 'react-image-lightbox'
+import { animated, useTransition } from 'react-spring'
+import { Box } from 'rebass/styled-components'
 
-import { StyledPanel } from 'components'
-import {
-  ImageZoomGrid,
-  ImageZoomGridElement,
-  PhotographySectionHeader,
-} from 'components/Photography'
+import { useDimensions, useMedia } from 'utils/hooks'
+
+import { ImageZoomGridElement, PhotographySectionHeader, SeeMoreLink } from '.'
 
 interface Props {
   datetime: DateTime
   // The images to be *currently* displayed for this section.
   images: any[]
-  // The *total* number of images available for this section, even those that
-  // are currently hidden.
-  // imageCount: number
-  // True if on `/photography` page; false if on one of the photo details pages.
-  isPreview?: boolean
   slug?: string
+  totalNumImages?: number
 }
 
-interface State {
-  index: number
-  isLightboxOpen: boolean
-  lightboxImages: string[]
-  lightboxSrc?: string
-  nextImageSrc?: string
-  prevImageSrc?: string
-}
+const PhotographyGridSection = (props: Props) => {
+  const { datetime, images, slug = '/#', totalNumImages = 0 } = props
+  const [isLightboxOpen, setIsLightboxOpen] = useState(false)
+  const [index, setLightboxIndex] = useState(0)
 
-interface ToggleLightboxOptions {
-  index: number
-  isLightboxOpen: boolean
-  lightboxImages?: string[]
-}
+  const sortedImages = _.sortBy(images, 'EXIF.DateTimeOriginal')
+  const lightboxImages = _.map(sortedImages, 'childImageSharp.sizes.srcWebp')
 
-class PhotographyGridSection extends Component<Props, State> {
-  public readonly state: State = {
-    index: 0,
-    isLightboxOpen: false,
-    lightboxImages: [],
+  const decrementLightboxIndex = () => setLightboxIndex(index - 1)
+  const incrementLightboxIndex = () => setLightboxIndex(index + 1)
+  const closeLightbox = () => setIsLightboxOpen(false)
+  const openLightbox = (imageIndex: number) => {
+    setLightboxIndex(imageIndex)
+    setIsLightboxOpen(true)
   }
-
-  /**
-   * Lightbox images are just the scaled up version of the thumbnails.
-   * Here we extract the absolute source path for all Lightbox images
-   * for the current gallery.
-   */
-  private getLightboxImagesFromProps: (props: Props) => string[] = _.memoize(
-    _.flow(
-      fp.get('images'),
-      fp.sortBy('EXIF.DateTimeOriginal'),
-      fp.map('childImageSharp.sizes.src'),
-    ),
+  const getImageAtIndex = (imageIndex: number) =>
+    _.get(lightboxImages, imageIndex % lightboxImages.length)
+
+  const lightboxSrc = getImageAtIndex(index)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Expected indentation of 4 spaces but found 2. (@typescript-eslint/indent)](https://app.codacy.com/app/jdstuart/jesses.io/pullRequest?prid=4576429)